src/market/directory.rs

```diff
@@ -315,32 +367,63 @@ pub fn start_directory_server(directory: Arc<DirectoryServer>) -> Result<(), Dir
 }
 
 // The stream should have read and write timeout set.
-// TODO: Use serde encoded data instead of string.
 fn handle_client(
     stream: &mut TcpStream,
     directory: &Arc<DirectoryServer>,
+    rpc: &Client,
 ) -> Result<(), DirectoryServerError> {
-    let reader_stream = stream.try_clone()?;
-    let mut reader = BufReader::new(reader_stream);
-    let mut request_line = String::new();
-
-    reader.read_line(&mut request_line)?;
-    if request_line.starts_with("POST") {
-        let addr: String = request_line.replace("POST ", "").trim().to_string();
-        directory.addresses.write()?.insert(addr.clone());
-        log::info!("Got new maker address: {}", addr);
-    } else if request_line.starts_with("GET") {
-        log::info!("Taker pinged the directory server");
-        let response = directory
-            .addresses
-            .read()?
-            .iter()
-            .fold(String::new(), |acc, addr| acc + addr + "\n");
-        stream.write_all(response.as_bytes())?;
-        stream.flush()?;
+    let buf = read_message(&mut stream.try_clone()?)?;
+    let value: DnsRequest = serde_cbor::de::from_reader(&buf[..])?;
+    match value {
+        DnsRequest::Post { metadata } => {
+            log::info!("Received new maker address: {}", &metadata.url);
+
+            let txid = metadata.proof.bond.outpoint.txid;
+            let transaction = rpc.get_raw_transaction(&txid, None)?;
+            let current_height = rpc.get_block_count()?;
+
+            match verify_fidelity_checks(
+                &metadata.proof,
+                &metadata.url,
+                transaction,
+                current_height,
+            ) {
+                Ok(_) => {
+                    log::info!("Maker verified successfully.");
+                    directory.addresses.write()?.insert(AddressEntry(
+                        metadata.proof.bond.amount.to_sat(),
+                        metadata.url.clone(),
+                    ));
+                }
+                Err(_) => {
+                    log::warn!("Potentially suspicious maker detected: {:?}", metadata.url);
```

Comment:
let's also log the error? Allows us to see what type of failure happened.

Reply:
Cool